Signal/src/ViewControllers/MediaGalleryViewController.swift

```diff
@@ -418,11 +423,11 @@ class MediaGalleryViewController: UINavigationController, MediaGalleryDataSource
         }
     }
 
-    public func dismissSelf(animated isAnimated: Bool, completion: (() -> Void)? = nil) {
+    public func dismissDetailView(_ pageViewController: MediaPageViewController, animated isAnimated: Bool, completion: (() -> Void)?) {
         self.view.isUserInteractionEnabled = false
         UIApplication.shared.isStatusBarHidden = false
 
-        guard let detailView = pageViewController?.view else {
+        guard let detailView = pageViewController.view else {
```

Comment:
So previously we were accessing a local, optional property named `pageViewController` and now we're accessing a non-optional, method parameter of the same name?  This is fine here, but when one identifier "shadows" another, its easy to write bugs.  I wonder if we could avoid that somehow, but don't have a proposal.

Reply:
You're right, I shouldn't have a method param that mirrors the name of a local property.

I think the only time we should have multiple symbols with the same name in a single scope is when unwrapping; e.g.

```
guard let foo = self.foo else {
   // ...
}
```

Since in that case there's no worry of referencing the wrong thing, and at worst you could get momentarily confused about the nullability of the thing, but the compiler ultimately ensures you get it right.

In this case, even though it should be fine, since the method param *happens*  to refer to the same instance as the local property, in general it's bad practice. Fixed!